tools/cell_census_builder/globals.py

```diff
@@ -216,11 +217,28 @@
 # Global TileDB context
 _TileDB_Ctx: tiledb.Ctx = None
 
+# The logical timestamp at which all builder data should be recorded
+WRITE_TIMESTAMP = int(time.time() * 1000)
+
+# Using "end of time" for read_timestamp means that all writes are visible, no matter what write timestamp was used
+END_OF_TIME = 0xFFFFFFFFFFFFFFFF
+
 
 def SOMA_TileDB_Context() -> soma.options.SOMATileDBContext:
     global _SOMA_TileDB_Context
     if _SOMA_TileDB_Context is None:
-        _SOMA_TileDB_Context = soma.options.SOMATileDBContext(tiledb_ctx=TileDB_Ctx())
+        # Set write timestamp to "now", so that we use consistent timestamps across all writes (mostly for aesthetic
+        # reasons). Set read timestamps to be same as write timestamp so that post-build validation reads can "see"
+        # the writes. Without setting read timestamp explicitly, the read timestamp would default to a time that
+        # prevents seeing the builder's writes.
+        _SOMA_TileDB_Context = soma.options.SOMATileDBContext(
+            tiledb_ctx=TileDB_Ctx(),
```

Comment:
what happens if `_TileDB_Ctx` is modified after the initial per-process call to `SOMA_TileDB_Context`?  It would not pick up the change.

I suggest you change line 229 to read:

> if _SOMA_TileDB_Context is None or _SOMA_TileDB_Context.tiledb_ctx != _TileDB_Ctx: